_posts/2022-03-16-solidity-0.8.13-release-announcement.md

```diff
@@ -0,0 +1,218 @@
+is primarily meant to be a straightforward transformation of the high-level Solidity code. This makes it more
+modular and easier to audit while the new Yul-based optimizer is powerful enough to remove any redundancies
+from the optimized version.
+
+The performance of the new pipeline is not yet always superior to the old one, but it
+can do much higher-level optimization across functions, so please try it out and give us feedback!
+
+One important feature of the new pipeline is that it can move stack variables to memory
+and thus avoid the "stack too deep" issue. In order to do this safely, the compiler needs to know
+that the memory slots it intends to use are not overwritten by inline assembly.
+Inline assembly is considered memory-safe if it only uses memory that has been previously allocated
+either by high-level Solidity code or by reading from the free memory pointer at ``0x40``.
+At the very beginning of the code, the compiler will set the free memory pointer to a position
+after the area reserved for the stack variables that were moved to memory.
+Since it is difficult to impossible for the compiler to check that inline assembly only accesses memory in a safe way,
+we added a syntactic mechanism for the developer to assert this:
+
+```
+assembly ("memory-safe") { ... }
+```
+
+Marking an unsafe block memory-safe can lead to undefined behavior 
+By default inline assembly blocks are not considered memory-safe and even a single such block disables the stack-to-memory mechanism for the whole contract.
+Assuming otherwise would be dangerous and due to the risk of stack variables moved to memory being overwritten.
+
+The module that can move stack variables to memory is only activated if all inline assembly
+blocks either do not access memory at all or are marked as being memory safe.
+
+If you are e.g. writing a code library that is meant to be used across compiler versions,
+you can also use a special comment syntax that has the same effect:
+/// @solidity memory-safe-assembly
+assembly { ... }
+
+We advise against using this syntax, though, as we also advise against using the same
+code base across breaking compiler versions.
+
+### `using for` at File Level with Global Binding
+
+This release extends the ``using for`` directive considerably. It can now be used
+at file level, it is possible to use it with free functions and
+it has the option to make its effect global. We believe that in combination
+with the recently released user-defined value types, this makes it possible
+to design proper reusable code libraries.
+
+In a future release, we plan to also add operators to user-defined types
+so that things like fixed-point type libraries are really easy to use
+and feel like compiler built-in types.
+
+If you use ``using {f, g, h} for Type;``, then the compiler
+verifies that the given functions can receive a ``Type`` value as first
```

Comment:
I'll rephrase that and remove the type checking comment, it's really not that relevant.